### PR TITLE
Alc 89

### PR DESCRIPTION
**Pull request by ledes**
Notes: 
- Currently adding tests
- Please be picky with my typescript, I need  to get more familiar  with it

Acceptance Criteria:

- [x] A user can navigate to a Fair Lunch Pool using: url/[balancer-pool-id]

- If the pool is a valid Fair Launch Pool the user can access the page. Matches the following criteria (AC 3 from ALC-87):

- [x] Finalized = false; this means this is a pool with at least 1 of 6 pool rights controlled by a smart contract

- [x] Only two tokens in token list
- [x]  Only one of the tokens in the pair must be WETH, DAI, or USDC (this reprsents the funding token)

- [ ] Currently have active gradual weights update underway. Ex: from April 15, 2021, 03:02 AM (12243100) until April 18, 2021, 02:32 AM (12262900)

-  [ ] Or have finished their gradual weights update within the last month

- [x] If the pool does not match Fair Launch criteria, we get a 404 Invalid Fair Launch ID (if better status then 404 available that also works)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -11,22 +12,32 @@ function App() {
       <Switch>
         <RouteWrapper
```
Comment (hosainnet) [inline src/App.tsx:13]: I have changed this btw if you want to pull again

Comment (ledes) [reply src/App.tsx]: updated  now :)

File: src/components/RouteWrapper/index.test.tsx [hunk 2]
```diff
@@ -1,52 +1,61 @@
...
 import { render, screen } from "@testing-library/react";
+import PoolDashboard from "../../pages/PoolDashboard";
 import PoolsList from "../../pages/Pools";
 import RouteWrapper from "../RouteWrapper";
 import NoMatch from "../../pages/NoMatch";
 import { BrowserRouter, Switch, MemoryRouter } from "react-router-dom";
 import { MockedProvider } from "@apollo/client/testing";
 
-test("Valid route is displayed", () => {
+test("Displayes the pools route", () => {
```
Comment (pygeek) [inline src/components/RouteWrapper/index.test.tsx:10]: This may be a typo.

**Review by pygeek — DISMISSED** [review]
Approved, but noted small typo

File: src/components/RouteWrapper/index.test.tsx (hunk 2 above)
Comment (ledes) [reply src/components/RouteWrapper/index.test.tsx]: Good catch!

File: src/components/RouteWrapper/index.tsx [hunk 3]
```diff
@@ -1,29 +1,51 @@
...
 interface Props extends RouteProps {
   component: React.ComponentType<any>;
-  requiresValidation: boolean;
+  requiresPoolValidation?: boolean;
 }
 
-const isRouteValid = () => {
-  // TODO function to validate  if the parameters are valid
+const isPoolUrlValid = (match: any) => {
+  if (!match || !match.params.poolId) {
```
Comment (moldovanflorin93) [inline src/components/RouteWrapper/index.tsx:14]:
```suggestion
  if (!match?.params?.poolId) {
```

File: src/components/RouteWrapper/index.tsx [hunk 4]
```diff
@@ -1,29 +1,51 @@
...
   ...rest
 }: Props) => {
-  if (requiresValidation && !isRouteValid()) {
-    return <Redirect to="/404" />;
+  if (requiresPoolValidation) {
+    const match: any = useRouteMatch("/pools/:poolId");
+    if (!isPoolUrlValid(match)) {
+      return <Redirect to="/404" />;
+    }
+    const poolId = match.params.poolId;
```
Comment (moldovanflorin93) [inline src/components/RouteWrapper/index.tsx:31]: const poolId = match?.params?.poolId;

File: src/components/RouteWrapper/index.tsx [hunk 5]
```diff
@@ -1,29 +1,51 @@
...
+    const { loading, error, data } = useQuery(GET_VALID_POOL_BY_ID, {
+      variables: { poolId: poolId },
+    });
+
+    if (loading) {
+      return <FullScreenLoading />;
+    }
+    if (
+      error ||
+      data.pools.length === 0 ||
```
Comment (moldovanflorin93) [inline src/components/RouteWrapper/index.tsx:41]:
```suggestion
      data?.pools?.length === 0 ||
```

**Review by moldovanflorin93 — COMMENTED** [review]
NAB: Optional Chaining
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html

File: src/components/RouteWrapper/index.tsx (hunk 5 above)
Comment (ledes) [reply src/components/RouteWrapper/index.tsx]: Done. Thanks for the feedback!

**Review by hosainnet — APPROVED** [review]

File: src/components/RouteWrapper/index.tsx (hunk 5 above)
Comment (moldovanflorin93) [reply src/components/RouteWrapper/index.tsx]: I hope I will can help a lot

**Pull request merged**